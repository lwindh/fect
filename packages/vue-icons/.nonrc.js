const { join } = require('path')

module.exports = {
  entry: join(__dirname, 'packages'),
  formats: 'noumd',
}
