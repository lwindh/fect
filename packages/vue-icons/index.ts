import { build } from './src/bundler'

//  iife ~
;(() => build())()
