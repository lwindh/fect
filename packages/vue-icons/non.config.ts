import path from 'path'
import { defineNonConfig } from '@fect-ui/cli'

export default defineNonConfig({
  entry: path.join(__dirname, 'packages'),
  formats: 'noumd',
  lib: {
    input: path.join(__dirname, 'packages'),
    format: 'noumd'
  }
})
