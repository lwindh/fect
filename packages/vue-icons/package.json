{
  "name": "@fect-ui/vue-icons",
  "version": "1.0.7",
  "author": "XeryYue",
  "description": "Well prepared high quality Icon Library for fect",
  "scripts": {
    "build": "ts-node index.ts && rm -rf svg && rm -rf packages"
  },
  "main": "lib/index.js",
  "module": "es/index.js",
  "types": "types/index.d.ts",
  "license": "MIT",
  "files": [
    "LISCENSE",
    "es/",
    "lib/",
    "types/",
    "README.md"
  ],
  "devDependencies": {
    "@fect-ui/cli": "1.0.0",
    "ts-node": "^10.3.0",
    "webfont": "^11.2.26",
    "jsdom": "^16.6.0",
    "@types/svgo": "1.3.2",
    "svgo": "1.3.2",
    "@types/jsdom": "^16.2.11",
    "ora": "5.4.1"
  },
  "peerDependencies": {
    "vue": "^3.0.11"
  },
  "homepage": "https://github.com/fay-org/fect",
  "repository": {
    "type": "git",
    "url": "https://github.com/fay-org/fect"
  }
}
