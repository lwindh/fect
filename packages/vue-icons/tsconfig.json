{
  "compilerOptions": {
    "baseUrl": ".",
    "outDir": "./lib",
    "target": "ES2015",
    "module": "CommonJS",
    "sourceMap": false,
    "declaration": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "importHelpers": true,
    "allowJs": true,
    "jsx": "preserve",
    "lib": ["esnext", "dom"]
  },
  "include": ["src/**/*"]
}
