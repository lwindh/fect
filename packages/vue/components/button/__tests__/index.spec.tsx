import Button from '..'
import { mount } from '@vue/test-utils'
import { github } from '@fect-ui/vue-icons'

describe('Button', () => {
  it('should be render as a element', () => {
    const wrapper = mount(Button, {
      slots: {
        default: 'Button',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(() => wrapper.unmount()).not.toThrow()
  })
  it('should be support diferent types', async () => {
    const wrapper = mount({
      data() {
        return { ghost: false }
      },
      template: `
      <div class="container">
        <fe-button type="default" :ghost="ghost">Button</fe-button>
        <fe-button type="success" :ghost="ghost">Button</fe-button>
        <fe-button type="warning" :ghost="ghost">Button</fe-button>
        <fe-button type="error" :ghost="ghost">Button</fe-button>
      </div>`,
      components: {
        [Button.name]: Button,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    await wrapper.setData({ ghost: true })
    expect(wrapper.findAll('.is-ghost').length).toEqual(4)
  })
  it('should be support different sizes', () => {
    const wrapper = mount({
      template: `
      <div class="container">
        <fe-button size="mini">Button</fe-button>
        <fe-button size="small">Button</fe-button>
        <fe-button size="medium">Button</fe-button>
        <fe-button size="large">Button</fe-button>
        <fe-button auto>Button</fe-button>
      </div>`,
      components: {
        [Button.name]: Button,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should be support loadType and loadColor follow button color', () => {
    const wrapper = mount({
      template: `
      <div class="container">
        <fe-button loading>Button</fe-button>
        <fe-button loading type="success" load-type="cube">Button</fe-button>
        <fe-button loading type="warning" load-type="wave">Button</fe-button>
      </div>`,
      components: {
        [Button.name]: Button,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.findAll('.loading__cube')).toBeTruthy()
    expect(wrapper.findAll('.loading__wave')).toBeTruthy()
  })
  it('should be allow some status ', () => {
    const wrapper = mount({
      template: `
      <div class="container">
        <fe-button disabled>Button</fe-button>
        <fe-button shadow>Button</fe-button>
      </div>`,
      components: {
        [Button.name]: Button,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should be support icon', () => {
    const wrapper = mount({
      data: () => ({
        icon: () => <github />,
      }),
      template: `
      <div class="container">
        <fe-button :icon="icon()">Button</fe-button>
      </div>`,
      components: {
        [Button.name]: Button,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should only emit event when button status not be loading or disabled', async () => {
    const wrapper = mount(Button)
    const el = wrapper.find('.fect-button')
    await el.trigger('click')
    expect(wrapper.emitted('click')).toBeTruthy()
    await wrapper.setProps({ disabled: true })
    await el.trigger('click')
    console.log(wrapper.emitted('click'))
    expect(wrapper.emitted('click'))
    await wrapper.setProps({ loading: true })
    await el.trigger('click')
    expect(wrapper.emitted('click'))
  })
})
