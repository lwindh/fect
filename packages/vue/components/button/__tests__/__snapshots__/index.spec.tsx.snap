// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should be allow some status  1`] = `
<div class="container"><button class="fect-button fect-button--default fect-button--medium is-disabled" style="--button-hover-bg: var(--primary-background); --button-hover-border: var(--primary-foreground); --button-hover-color: var(--primary-foreground);" type="button">
    <!---->
    <!---->Button
  </button><button class="fect-button fect-button--default fect-button--medium is-shadow" style="--button-hover-bg: var(--primary-background); --button-hover-border: var(--primary-foreground); --button-hover-color: var(--primary-foreground);" type="button">
    <!---->
    <!---->Button
  </button></div>
`;

exports[`Button should be render as a element 1`] = `
<button class="fect-button fect-button--default fect-button--medium " style="--button-hover-bg: var(--primary-background); --button-hover-border: var(--primary-foreground); --button-hover-color: var(--primary-foreground);" type="button">
  <!---->
  <!---->Button
</button>
`;

exports[`Button should be support diferent types 1`] = `
<div class="container"><button class="fect-button fect-button--default fect-button--medium " style="--button-hover-bg: var(--primary-background); --button-hover-border: var(--primary-foreground); --button-hover-color: var(--primary-foreground);" type="button">
    <!---->
    <!---->Button
  </button><button class="fect-button fect-button--success fect-button--medium " style="--button-hover-bg: var(--primary-background); --button-hover-border: var(--success-default); --button-hover-color: var(--success-default);" type="button">
    <!---->
    <!---->Button
  </button><button class="fect-button fect-button--warning fect-button--medium " style="--button-hover-bg: var(--primary-background); --button-hover-border: var(--warning-default); --button-hover-color: var(--warning-default);" type="button">
    <!---->
    <!---->Button
  </button><button class="fect-button fect-button--error fect-button--medium " style="--button-hover-bg: var(--primary-background); --button-hover-border: var(--error-default); --button-hover-color: var(--error-default);" type="button">
    <!---->
    <!---->Button
  </button></div>
`;

exports[`Button should be support different sizes 1`] = `
<div class="container"><button class="fect-button fect-button--default fect-button--mini " style="--button-hover-bg: var(--primary-background); --button-hover-border: var(--primary-foreground); --button-hover-color: var(--primary-foreground);" type="button">
    <!---->
    <!---->Button
  </button><button class="fect-button fect-button--default fect-button--small " style="--button-hover-bg: var(--primary-background); --button-hover-border: var(--primary-foreground); --button-hover-color: var(--primary-foreground);" type="button">
    <!---->
    <!---->Button
  </button><button class="fect-button fect-button--default fect-button--medium " style="--button-hover-bg: var(--primary-background); --button-hover-border: var(--primary-foreground); --button-hover-color: var(--primary-foreground);" type="button">
    <!---->
    <!---->Button
  </button><button class="fect-button fect-button--default fect-button--large " style="--button-hover-bg: var(--primary-background); --button-hover-border: var(--primary-foreground); --button-hover-color: var(--primary-foreground);" type="button">
    <!---->
    <!---->Button
  </button><button class="fect-button fect-button--default fect-button--medium is-auto" style="--button-hover-bg: var(--primary-background); --button-hover-border: var(--primary-foreground); --button-hover-color: var(--primary-foreground);" type="button">
    <!---->
    <!---->Button
  </button></div>
`;

exports[`Button should be support icon 1`] = `
<div class="container"><button class="fect-button fect-button--default fect-button--medium " style="--button-hover-bg: var(--primary-background); --button-hover-border: var(--primary-foreground); --button-hover-color: var(--primary-foreground);" type="button">
    <!---->
    <!---->
    <div class="fect-button__icon" style="left: 15%; transform: translateY(-50%);"><svg fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5" shape-rendering="geometricPrecision" viewBox="0 0 24 24" height="24" width="24" style="color: currentColor;">
        <path d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 00-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0020 4.77 5.07 5.07 0 0019.91 1S18.73.65 16 2.48a13.38 13.38 0 00-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 005 4.77a5.44 5.44 0 00-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 009 18.13V22"></path>
      </svg></div>
    <div class="fect-button__text"> Button </div>
  </button></div>
`;

exports[`Button should be support loadType and loadColor follow button color 1`] = `
<div class="container"><button class="fect-button fect-button--default fect-button--medium is-loading" style="--button-hover-bg: var(--primary-background); --button-hover-border: var(--primary-foreground); --button-hover-color: var(--primary-foreground);" type="button">
    <div class="fect-button__loading">
      <div class="fect-loading"><span class="loading"><i class="loading__default default medium"></i><i class="loading__default default medium"></i><i class="loading__default default medium"></i></span></div>
    </div>
    <!---->Button
  </button><button class="fect-button fect-button--success fect-button--medium is-loading" style="--button-hover-bg: var(--primary-background); --button-hover-border: var(--success-default); --button-hover-color: var(--success-default);" type="button">
    <div class="fect-button__loading">
      <div class="fect-loading"><span class="loading"><i class="loading__cube default medium"></i><i class="loading__cube default medium"></i><i class="loading__cube default medium"></i><i class="loading__cube default medium"></i></span></div>
    </div>
    <!---->Button
  </button><button class="fect-button fect-button--warning fect-button--medium is-loading" style="--button-hover-bg: var(--primary-background); --button-hover-border: var(--warning-default); --button-hover-color: var(--warning-default);" type="button">
    <div class="fect-button__loading">
      <div class="fect-loading"><span class="loading"><i class="loading__wave default medium"></i><i class="loading__wave default medium"></i><i class="loading__wave default medium"></i><i class="loading__wave default medium"></i><i class="loading__wave default medium"></i></span></div>
    </div>
    <!---->Button
  </button></div>
`;
